syntax = "proto3";
package pt.ulisboa.tecnico.tuplespaces.centralized.contract;

message ServerAddress {
  string host = 1;
  int32 port = 2;
}

message RegisterRequest {
  string serviceName = 1;
  string qualifier = 2;
  ServerAddress address = 3;
}

message RegisterResponse {
}


message LookupServerRequest {
  string serviceName = 1;
  string qualifier = 2;
}

message ServerInfo {
  ServerAddress address = 1;
  string qualifier = 2;
}

message LookupServerResponse {
  repeated ServerInfo serverInfo = 1;
}

message DeleteServerRequest {
  string serviceName = 1;
  ServerAddress address = 2;
}

message DeleteServerResponse {}

service NameServerService {
  rpc register(RegisterRequest) returns (RegisterResponse);
  rpc lookup(LookupServerRequest) returns (LookupServerResponse);
  rpc delete(DeleteServerRequest) returns (DeleteServerResponse);
}
